curve25519-dalek/src/edwards.rs

```diff
@@ -574,6 +575,26 @@ impl EdwardsPoint {
         CompressedEdwardsY(s)
     }
 
+    /// Compress several `EdwardsPoint`s into `CompressedEdwardsY`.
+    #[cfg(feature = "alloc")]
+    pub fn compress_batch(inputs: &[EdwardsPoint]) -> Vec<CompressedEdwardsY> {
```

Comment:
Note: this and `FieldElement::batch_invert` could avoid the `alloc` requirement by using const generic-sized arrays, which allows storing intermediates in a same-sized array rather than a `Vec`.

In the @RustCrypto crates we support both a no-alloc const generic and alloc-backed slice-based inputs for these sorts of APIs.

Reply:
Agreed we could probably very easily provide this (assuming we're okay exposing a generic array crate, which I don't think we're doing now)

Reply:
You don't need a generic array crate. It works fine with existing const generics, e.g.:

```rust
fn batch_invert<const N: usize>(elems: &mut [Self; N]) {
    let scratch = [Self::ONE; N];
    ...
}
```

Functions like that can infer the stack size needed for temporaries based on the size of the input, avoiding a heap allocation